core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -164,29 +171,38 @@ object Parse {
 
     val attrs = strings.drop(1).map({ x => {
       if (x.mkString.contains(argSeparator)) {
-        throw new ModuleParseError(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
+        return Left(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
           s"'org${argSeparator}name[${argSeparator}version][${argSeparator}config]${attrSeparator}attr1=val1${attrSeparator}attr2=val2'")
       }
       val y = x.split("=")
       if (y.length != 2) {
-        throw new ModuleParseError(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
+        return Left(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
       }
       (y(0), y(1))
     }
     }).toMap
 
+    // Only "classifier" and "url" attributes are allowed
+    val validAttrsKeys = Set("classifier", "url")
+
+    val maybeErrorMsg = validateAttributes(attrs, validAttrsKeys)
```

Comment:
Instead of the `.isDefined` / ` return Left(…)`, we can do something like
```scala
validateAttributes(attrs, validAttrsKeys) match {
  case Some(err) => Left(err)
  case None =>
    …
}
```
which uses neither `.get` nor `return` (`.get` is ~unsafe, as it can throw exceptions if ever we forgot it to ensure it's a `Some(…)`, and `return` has [many drawbacks](https://tpolecat.github.io/2014/05/09/return.html) in Scala).